src/pull_request.ts

```diff
@@ -53,14 +47,52 @@ export async function handlePullRequest() {
   let overviewComment = existingComments.find((comment) =>
     comment.body?.includes(OVERVIEW_MESSAGE_SIGNATURE)
   );
+  const isIncrementalReview = !!overviewComment;
+
+  // Maybe fetch review comments
+  const reviewComments = isIncrementalReview
+    ? (
+        await octokit.rest.pulls.listReviewComments({
+          ...context.repo,
+          pull_number: pull_request.number,
+        })
+      ).data
+    : [];
+
+  // Get modified files
+  const { data: files } = await octokit.rest.pulls.listFiles({
+    ...context.repo,
+    pull_number: pull_request.number,
+  });
+  const fileDiffs = files.map((file) => parseFileDiff(file, reviewComments));
+  info(`successfully fetched file diffs`);
+
+  let commitsReviewed: string[] = [];
   if (overviewComment) {
+    info(`running incremental review`);
+    try {
+      const payload = JSON.parse(
+        overviewComment.body
+          ?.split(PAYLOAD_TAG_OPEN)[1]
+          .split(PAYLOAD_TAG_CLOSE)[0] || "{}"
+      );
```

Comment:
The current implementation might throw errors if `PAYLOAD_TAG_OPEN` or `PAYLOAD_TAG_CLOSE` are not found in the string, as array indexing with [1] and [0] could return undefined. Consider using array destructuring with default values or additional checks to make this more robust:
```typescript
const parts = overviewComment.body?.split(PAYLOAD_TAG_OPEN) || [];
const payloadStr = parts[1]?.split(PAYLOAD_TAG_CLOSE)[0] || "{}";
const payload = JSON.parse(payloadStr);
```